js/index.js

```diff
@@ -0,0 +1,210 @@
+        this.eventRegist();
+    }
+
+    getTodoEntity(idx) {
+        let $todoItem = this.$todoList.querySelector(`[data-idx='${idx}']`);
+        return {
+            idx:Number(idx),
+            value:$todoItem.querySelector('.label').textContent,
+            check:$todoItem.querySelector('.toggle').checked
+        }
+    }
+
+    getLastIndex() {
+        let todoItems = lsg.get('todoItems');
+        let max = 0;
+        if (!todoItems || todoItems.length === 0) return max;
+
+        for (const item of todoItems) {
+            if (max < item.idx) max = item.idx;
+        }
+        return max;
+    }
+
+    getNextIndex() {
+        return this.getLastIndex() + 1;
+    }
+
+    save = (entity) => {
+        let todoItems = lsg.get("todoItems");
+        if (!todoItems) todoItems = [];
+        let findItem = todoItems.find(a => a.idx === entity.idx);
+        if (findItem) {
+            findItem.value = entity.value;
+            findItem.check = entity.check;
+        } else {
+            todoItems.push(entity);
+        }
+        lsg.set('todoItems', todoItems);
+    }
+    delete = (idx) => {
+        if(whatType(idx)==="string")idx = Number(idx);
+        let todoItems = lsg.get("todoItems");
+        let newTodoItems = [];
+        if (todoItems.length > 0) {
+            for (const todoItem of todoItems) {
+                if (todoItem.idx !== idx) {
+                    newTodoItems.push(todoItem);
+                }
+            }
+        }
+        lsg.set('todoItems', newTodoItems);
+    }
+
+    //LocalStorage확인 후 Load하는 함수
+    build = () => {
+        let todoItems = lsg.get("todoItems");
+        if (todoItems.length) {
+            for (const item of todoItems) {
+                this.$todoList.append(this.getFrame(item.idx, item.value, item.check));
+            }
+            this.updateCount();
+        }
+    }
+
+    //이벤트 등록
+    eventRegist = () => {
+        this.$todoInput.addEventListener('keyup', this.onAddInput);
+        this.$todoFooter.addEventListener('click', this.switchTap);
+        this.$todoList.addEventListener('click', this.onChangeToggle);
+        this.$todoList.addEventListener('dblclick', this.onEditingByDblClick);
+        this.$todoList.addEventListener('keyup', this.onEditingByKeyup);
+    }
+
+    //수정적용 및 취소
+    onEditingByKeyup = ({ target, key }) => {
+        if (!target) return;
+        const close = () => target.parentElement.classList.remove('editing');
+        if (target.tagName === "INPUT" && key === "Escape") {
+            target.parentElement.classList.remove('editing');
+            target.value = this.previous;
+            close();
+        } else if (target.tagName === "INPUT" && key === "Enter") {
+            //TODO 공백으로 만든 뒤 엔터를 칠 경우의 로직 구현 필요
+            target.previousElementSibling.querySelector(".label").textContent = target.value;
+            this.save(this.getTodoEntity(target.parentElement.dataset["idx"]));
+            close();
+        }
```

Comment:
```suggestion
        if (target.tagName === "INPUT" && key === "Enter") {
            //TODO 공백으로 만든 뒤 엔터를 칠 경우의 로직 구현 필요
            target.previousElementSibling.querySelector(".label").textContent = target.value;
            this.save(this.getTodoEntity(target.parentElement.dataset["idx"]));
        }
        if (target.tagName === "INPUT" && ["Escape", "Enter"].includes(key)) {
            close();
        }
```
이렇게 enter를 먼저 처리하면 중복코드를 제거할 수 있습니다.